cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -880,4 +968,37 @@ private MetadataScope validateScope(String scope) throws BadRequestException {
                                                   scope, MetadataScope.USER, MetadataScope.SYSTEM));
     }
   }
+
+  private NavigableSet<MetadataSearchResultRecord> applySorting(Set<MetadataSearchResultRecord> original, String sortBy,
+                                                                String sortOrder) throws BadRequestException {
+    Comparator<MetadataSearchResultRecord> comparator = getComparator(sortBy);
+    TreeSet<MetadataSearchResultRecord> result = new TreeSet<>(comparator);
+    result.addAll(original);
+    if ("desc".equalsIgnoreCase(sortOrder)) {
+      return result.descendingSet();
+    }
+    return result;
+  }
+
+  private Comparator<MetadataSearchResultRecord> getComparator(String sortBy) throws BadRequestException {
+    switch (sortBy) {
+      case "name":
+        return NAME_ASCENDING_COMPARATOR;
+      default:
```

Comment:
not yet, but it will be added after https://issues.cask.co/browse/CDAP-5438 is done.